### PR TITLE
[WIP] New dialogs

### PR DESCRIPTION
**Pull request by zbettenbuk**

### CODE REVIEW

File: react/features/recording/components/Recording/StartRecordingDialogContent.js [hunk 1]
```diff
@@ -52,6 +52,11 @@ type Props = {
      */
     t: Function,
 
+    /**
+     *  The color theme to be used (native only).
+     */
+    theme: ?string,
```
Comment (saghul) [inline react/features/recording/components/Recording/StartRecordingDialogContent.js]: why the inconsistency? shouldn't all dialogs be rendered equal? if theming support is required, can we extract it to its own commit then?

File: react/features/room-lock/components/PasswordRequiredPrompt.native.js [hunk 2]
```diff
@@ -62,12 +62,11 @@ class PasswordRequiredPrompt extends Component<Props> {
...
         return (
-            <Dialog
-                bodyKey = 'dialog.passwordLabel'
+            <InputDialog
+                contentKey = 'dialog.passwordLabel'
                 onCancel = { this._onCancel }
                 onSubmit = { this._onSubmit }
-                textInputProps = { _TEXT_INPUT_PROPS }
-                titleKey = 'dialog.passwordRequired' />
+                textInputProps = { _TEXT_INPUT_PROPS } />
```
Comment (saghul) [inline react/features/room-lock/components/PasswordRequiredPrompt.native.js]: Is this constant used elsewhere? If not, can we inline it since we're here?

File: react/features/feedback/components/styles.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+// @flow
+import { BoxModel, ColorPalette, createStyleSheet } from '../../base/styles';
```
Comment (saghul) [inline react/features/feedback/components/styles.js]: style: I think we leave a newline here

File: react/features/feedback/components/styles.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+export default createStyleSheet({
+    dialogComponent: {
+        paddingHorizontal: BoxModel.padding * 2,
+        paddingVertical: BoxModel.padding
+    },
+
+    feedbackHeader: {
+        minWidth: 300,
```
Comment (saghul) [inline react/features/feedback/components/styles.js]: I've seen this number before in this PR. It needs a constant I think. But more importantly, what about tablets? let's talk about it.

**Comment by saghul** [comment]
Some UI / UX feedback. Pinging @yanas too.

- Tapping out of the dialog doesn't cancel it. That was the previous behavior and I'd say it's the expectation.

On a more subjective tangent...

- I strongly dislike the X for closing dialogs. None of the Android or iOS *system* dialogs have it.
- I think we shoud use titles on dialogs, they look quite base otherwise.

Man do I love the black!

File: react/features/base/dialog/components/native/BaseDialog.js [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+export type Props = {
+    ...AbstractProps,
+
+    /**
+     * Overrides the overlay color.
+     *
+     * NOTE: This is defined in style by default, but some custom dialogs may
+     * need different color or opacity.
+     */
+    overlayOverride?: string,
```
[the remark this answers is not in the document]
Comment (zbettenbuk) [reply react/features/base/dialog/components/native/BaseDialog.js]: Good call, thanks!

File: react/features/base/dialog/components/native/BaseSubmitDialog.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+                        <Text style = { brandedDialog.text }>
+                            { t(this.props.okTitleKey || 'dialog.Ok') }
+                        </Text>
+                    </TouchableOpacity>
+                </View>
+            </View>
+        );
+    }
+
+    _onCancel: () => void;
```
[the remark this answers is not in the document]
Comment (zbettenbuk) [reply react/features/base/dialog/components/native/BaseSubmitDialog.js]: Yep :( Also, I like them, because it gives a strict format.

File: react/features/base/dialog/components/native/ConfirmDialog.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+import { brandedDialog } from './styles';
+
+type Props = {
+    ...BaseProps,
+
+    /**
+     * Untranslated i18n key of the content to be displayed.
+     *
+     * NOTE: This dialog also adds support to Object type keys that will be
+     * translated using the provided params. See {@code t(string, Object)} for
```
[the remark this answers is not in the document]
Comment (zbettenbuk) [reply react/features/base/dialog/components/native/ConfirmDialog.js]: Yeah, the actual function is called ```translate```, not ```t```. Fixed, thanks!

File: react/features/base/dialog/components/native/InputDialog.js [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+
+import { translate } from '../../../i18n';
+
+import { type State as AbstractState } from '../AbstractDialog';
+
+import BaseDialog, { type Props as BaseProps } from './BaseDialog';
+import {
+    brandedDialog,
+    inputDialog as styles,
+    FIELD_UNDERLINE
```
[the remark this answers is not in the document]
Comment (zbettenbuk) [reply react/features/base/dialog/components/native/InputDialog.js]: Can be 👨‍🍳

File: react/features/base/dialog/components/native/InputDialog.js [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+} from './styles';
+
+type Props = {
+    ...BaseProps,
+
+    contentKey: string,
```
[the remark this answers is not in the document]
Comment (zbettenbuk) [reply react/features/base/dialog/components/native/InputDialog.js]: It's written with invisible ink. But nevermind, I make it rgb(36, 41, 46) ;)

File: react/features/base/dialog/components/native/InputDialog.js [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+
+    t: Function,
+
+    textInputProps: ?Object
+}
+
+type State = {
+    ...AbstractState,
+
+    fieldValue: ?string
```
[the remark this answers is not in the document]
Comment (zbettenbuk) [reply react/features/base/dialog/components/native/InputDialog.js]: Same here

File: react/features/base/dialog/components/native/InputDialog.js [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+};
+
+/**
+ * Implements a single field input dialog component.
+ */
+class InputDialog extends BaseDialog<Props, State> {
```
[the remark this answers is not in the document]
Comment (zbettenbuk) [reply react/features/base/dialog/components/native/InputDialog.js]: These dialogs handle the ```okDisabled``` prop to achieve that functionality, same as the old Dialogs did. Although, I forgot to add it to this dialog, so thanks for pointing it out!

File: react/features/base/dialog/components/native/SubmitDialog.js [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+type Props = {
+    ...BaseProps,
+
+    t: Function
+}
+
+/**
+ * Implements a submit dialog component that can have free content.
+ */
+class SubmitDialog extends BaseSubmitDialog<Props, *> {
```
[the remark this answers is not in the document]
Comment (zbettenbuk) [reply react/features/base/dialog/components/native/SubmitDialog.js]: Good point, thanks

File: react/features/settings/DeviceSelectionPopup.js [hunk 13]
```diff
@@ -15,7 +15,7 @@ import { DeviceSelection } from '../device-selection';
 
 // Using the full path to the file to prevent adding unnecessary code into the
 // dialog popup bundle.
-import DialogWithTabs from '../base/dialog/components/DialogWithTabs';
+import { DialogWithTabs } from '../base/dialog';
```
[the remark this answers is not in the document]
Comment (zbettenbuk) [reply react/features/settings/DeviceSelectionPopup.js]: Nope, thanks!

File: react/features/authentication/components/LoginDialog.native.js [hunk 14]
```diff
@@ -163,36 +168,44 @@ class LoginDialog extends Component<Props, State> {
...
-                    <Text style = { styles.dialogText }>
-                        {
-                            messageKey
-                                ? t(messageKey, messageOptions)
-                                : connecting
-                                    ? t('connection.CONNECTING')
-                                    : ''
-                        }
-                    </Text>
+                    { (messageKey || connecting) && (
```
[the remark this answers is not in the document]
Comment (zbettenbuk) [reply react/features/authentication/components/LoginDialog.native.js]: Sure!

File: react/features/recording/components/Recording/StartRecordingDialogContent.js (hunk 1 above)
Comment (zbettenbuk) [reply react/features/recording/components/Recording/StartRecordingDialogContent.js]: Actually, that was intended for backward compatibility for web, but then I just found out that it's not really needed, so removed it. Thanks!

File: react/features/feedback/components/styles.js (hunk 4 above)
Comment (zbettenbuk) [reply react/features/feedback/components/styles.js]: I agree about the constants, did it. But also added a note to the constant to explain why it makes sense

**Comment by saghul** [comment]
NOTE: merged with a known and unrelated failing test.
